### PR TITLE
handle timeout error on sum_bucket/2

### PR DESCRIPTION
**Pull request by kuenishi**
addresses #743.
from commit message:

```
add error handling when sum_bucket failed with timeout, which
causes dirty error message and no way to know which bucket was
failed in usage calculation mapreduce. This commit adds two ways
to know that:

- outputs an error log that prints user key id and bucket name
  with error reason.
- stores the result into moss.storage with reason like:
    { "bucket_name" : "{error,{timeout,[]}}" }

The latter log proves and indicated there existed a bucket at
that time even though its usage size was unknown. Better than
ignored, which operator can't tell non-existent from calculation
failure.
```

### CODE REVIEW

**Comment by andrewjstone** [comment]
Code looks good. 
- [x] cs743_regression_test riak_test
- [x] eunit + eqc
- [x] dialyzer
- [x] xref

**Comment by andrewjstone** [comment]
I'm seeing the following dialyzer errors related to this change:

```
riak_cs_storage.erl:60: Invalid type specification for function riak_cs_storage:maybe_sum_bucket/3. The success typing is (pid(),[binary() | maybe_improper_list(any(),binary() | []) | byte()],binary() | [byte()]) -> {binary() | [byte()],binary()}
riak_cs_storage.erl:64: The pattern {'ok', BucketUsage} can never match the type {'error',_} | {'struct',[{_,_},...]}
```

**Comment by kuenishi** [comment]
Hooray! It was totally a bug found by dialyzer.
This regression could be found if #703 were done.

**Comment by andrewjstone** [comment]
:+1:

File: src/riak_cs_storage.erl [hunk 1]
```diff
@@ -46,12 +46,30 @@ sum_user(Riak, User) when is_binary(User) ->
...
-                   || ?RCS_BUCKET{name=B} <- Buckets ]};
+            BucketUsages = [maybe_sum_bucket(Riak, User, B)
+                            || ?RCS_BUCKET{name=B} <- Buckets],
+            {ok, BucketUsages};
         {error, Error} ->
             {error, Error}
     end.
 
+%% @doc Output a log when calculating total usage of a bucket.
+%%      This log is *very* important because unless this log
```
Comment (reiddraper) [inline src/riak_cs_storage.erl:57]: s/unless/without

Comment (kuenishi) [inline src/riak_cs_storage.erl:57]: Oh crap. Thank you for pointing it out.

**Pull request merged**